### PR TITLE
Added node deletion from BST #4301

### PR DESCRIPTION
**Pull request by Amulya310**
Solution to  #issue4301

Hey, I've added the code for node deletion from BST in CPP. This is a small contribution under GSSOC'21.
I've also included a brief description ,multiline comments and 3 test cases( inputs and outputs ) in my code.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
yes
(Write your answer here.)

### Description
This is a cpp code to delete a node from a binary search tree .The code takes into consideration all the possible cases that may arise .
(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#4301 
(Write your answer here.)

### CODE REVIEW

File: node_deletion_BST.cpp [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+		root->r = deleteNode(root->r, temp->key);
+	}
+	return root;
+}
+
+// Driver Code
+int main()
+{
+	struct node* root = NULL;
+	root = insert(root, 1);
```
Comment (atarax665) [inline node_deletion_BST.cpp:141]: Try to take inputs from the user.

File: node_deletion_BST.cpp [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+	cout << "\nDelete 14\n";
+	root = deleteNode(root, 14);
+	cout << "Inorder traversal of the modified tree \n";
+	inorder(root);
+
+	return 0;
+}
+
+/*test cases-
+1.input-2
```
Comment (atarax665) [inline node_deletion_BST.cpp:170]: Add space and time complexity of the program.

**Review by atarax665 — CHANGES_REQUESTED** [review]
Update appropriate README.

**Review by ankitaggarwal23 — COMMENTED** [review]
Code plagiarized from [here](https://github.com/tanvi-surana/data-structures/blob/master/trees/BinarySearchTree%20class1.cpp).

**Pull request closed without merging**